server-api/routes/products.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express');
+const product = require('../models/product');
+const router = express.Router();
+
+//Getting All
+router.get('/', async (req, res) => {
+    try {
+        const products = await product.find();
+        res.json(products);
+    } catch (err) {
+        res.status(500).json({message: err.message})
+    }
+});
+
+//Getting One using ID
+router.get('/:id', getProduct, (req, res) => {
+    // res.send(res.ProductById.name)
+    res.json(res.ProductById);
+});
+
+//Creating One
+router.post('/', async (req, res) => {
+    const ProductEntry = new product({
+        id: req.body.id,
+        name: req.body.name,
+        imageurl: req.body.imageurl,
+        coll: req.body.coll,
+        category: req.body.category,
+        price: req.body.price,
+        description: req.body.description,
+        rating: req.body.rating,
```

Comment:
use destructuring to fetch these values from the req.body - instead of repeating req.body for each value